src/main/java/ua/lviv/iot/parkingServer/model/Vehicle.kt

```diff
@@ -1,24 +1,17 @@
 package ua.lviv.iot.parkingServer.model
 
 import java.time.Duration
+import org.bson.types.ObjectId
+import org.springframework.data.annotation.Id
+import org.springframework.data.mongodb.core.mapping.Document
 import ua.lviv.iot.parkingServer.model.enums.VehicleType
 
+@Document("vehicle")
 data class Vehicle(
-    override var id: Long,
-    var number: String,
-    var typeOfVehicle: VehicleType,
-    var durationOfUseOfParkingSpot: Duration,
-    var isTicketReceived: Boolean
-) : CsvData {
-
-    override fun getHeaders(): String =
-        listOf(
-            "Id",
-            "Number",
-            "Vehicle Type",
-            " Duration of use parking spot",
-            "Is ticket received"
-        ).joinToString(separator = ", ")
-
-    override fun toCSV(): String = "$id, $number, $typeOfVehicle, $durationOfUseOfParkingSpot, $isTicketReceived"
-}
+    @Id
+    var id: String = ObjectId.get().toString(),
```

Comment:
Let's delegate id population to mongo driver.

Let's wait for @OlehSarapuk what he would say :)

Reply:
you'r right
(you can omit declaring id
just do it lateinit , mongo will initialize it for you)

Reply:
✅